clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -1234,8 +1245,8 @@ protected void resetEpoch() {
     protected void updateMemberEpoch(int newEpoch) {
         boolean newEpochReceived = this.memberEpoch != newEpoch;
         this.memberEpoch = newEpoch;
-        // Simply notify based on epoch change only, given that the member will never receive a
-        // new member ID without an epoch (member ID is only assigned when it joins the group).
+        // Simply notify based on epoch changes only, since the member will generate a member ID
+        // at startup, and it will remain unchanged for its entire lifetime.
         if (newEpochReceived) {
```

Comment:
I'm not sure if we need to update the logic here accordingly.

https://github.com/apache/kafka/blob/2267518d5bed60e2a1731712ae7cb455ea0bf8c4/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java#L1239-L1245

Reply:
Thanks for feedback. 😺 
I’m wondering about the initial value of `MemberInfo.memberId` if we replace it with a `String` instead of `Optional<String>`. Should we use an empty string as its initial value ?